### PR TITLE
Tempus fix imex initialization

### LINKED ISSUE #6862: Tempus:  IMEX Not Fully Constructed Through StepperFactory
## Bug Report
@trilinos/tempus 

### Description
When an IMEX stepper is constructed through the StepperFactory, it is not fully constructed so that the tableaus can be set later in the code.

Need to get the IMEX stepper fully constructed, an dneed to have some test to cover this.  Should survey other steppers and ensure that this is not happening with other steppers.

### PR DESCRIPTION
**Pull request by krcb**
@trilinos/tempus 

## Motivation

Addresses IMEX setup issues reported in #6862. @ccober6 you may want to see if this kind of approach needs to be applied in other places and what testing is needed.

## Related Issues

Closes #6862

### CODE REVIEW

**Comment by ccober6** [comment]
@krcb Just pushed some changes.  Look them over, and see what you think.  I have several meetings this after, so I will likely touch base with you tomorrow morning.

**Comment by krcb** [comment]
> @krcb Just pushed some changes. Look them over, and see what you think. I have several meetings this after, so I will likely touch base with you tomorrow morning.

@ccober6 Can we have versions of these tests that create the stepper with both a null parameter list and a null model?

**Comment by ccober6** [comment]
Sure.  Would a test that does the following sequence cover your case?
* Constructs the IMEX Stepper with a null parameterlist and null model
* Sets the model
* Test if the IMEX Stepper is correctly initialized.

Thus the parameterlist is the default parameterlist, or would you like the parameterlist set also?  It would be a little harder to test the state of the IMEX Stepper when the model is not set, but doable.

**Comment by ccober6** [comment]
@krcb @rppawlo Roger, can you review this?

**Comment by krcb** [comment]
@ccober6 The test changes look good. I can't approve though, as I opened the PR. @rppawlo can you review?

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 1]
```diff
@@ -1459,56 +1496,58 @@ class StepperFactory
...
+      while ((pos != std::string::npos) || (lastPos != std::string::npos)) {
+        std::string token = str.substr(lastPos,pos-lastPos);
+        // Strip single quotes
+        std::string::size_type beg = token.find_first_of("'") + 1;
+        std::string::size_type end = token.find_last_of ("'");
+        stepperListStr.push_back(token.substr(beg,end-beg));
+
+        lastPos = str.find_first_not_of(delimiters, pos); // Skip delimiters
+        pos = str.find_first_of(delimiters, lastPos);     // Find next delimiter
+      }
```
Comment (rppawlo) [inline packages/tempus/src/Tempus_StepperFactory.hpp]: Not saying we need to do this now, but I have similar code in panzer. One day we should move things like this into a teuchos string utils library.

File: packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp [hunk 2]
```diff
@@ -40,6 +40,8 @@ using Tempus::StepperExplicitRK;
 // Comment out any of the following tests to exclude from build/run.
 #define CONSTRUCTION
 #define STEPPERFACTORY_CONSTRUCTION
+#define STEPPERFACTORY_CONSTRUCTION_GENERAL_WO_PARAMETERLIST
+#define STEPPERFACTORY_CONSTRUCTION_GENERAL_WO_PARAMETERLIST_MODEL
```
Comment (rppawlo) [inline packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp]: Why do these defines exist? There is a runtime flag that you can use to selectively execute just a single test from an executable. Maybe this is more convenient?

File: packages/tempus/src/Tempus_StepperFactory.hpp [hunk 3]
```diff
@@ -1459,56 +1496,58 @@ class StepperFactory
...
-    using Teuchos::RCP;
-    using Teuchos::ParameterList;
+    if (stepperPL != Teuchos::null) {
+      using Teuchos::RCP;
+      using Teuchos::ParameterList;
+
+      // Parse Stepper List String
+      std::vector<std::string> stepperListStr;
+      stepperListStr.clear();
+      std::string str = stepperPL->get<std::string>("Stepper List");
```
[the remark this answers is not in the document]
Comment (ccober6) [reply packages/tempus/src/Tempus_StepperFactory.hpp]: No I do not think so.  I will remove it in another branch, as it does not hurt everything here.

File: packages/tempus/src/Tempus_StepperFactory.hpp (hunk 1 above)
Comment (ccober6) [reply packages/tempus/src/Tempus_StepperFactory.hpp]: Yeah, you shared it with me originally from panzer.  Putting it in teuchos would be good.

File: packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp (hunk 2 above)
Comment (ccober6) [reply packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp]: Ah, what does that command look like?  This original came from Irina, and I thought it was useful during debugging.

**Comment by krcb** [comment]
@ccober6 @rppawlo looks like this timed out waiting for a job and so testing failed. Do we need to do anything to restart this?

**Comment by ccober6** [comment]
I added the labels AUTOMERGE and RETEST.  Hopefully this will get it going again.

File: packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp (hunk 2 above)
Comment (rppawlo) [reply packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp]: test.exe --test=<my_test_name>

For example, I have a panzer unit test declared with:
```
  TEUCHOS_UNIT_TEST(global_data, builder)
  {
   ...
  }
```

You could run only that one with:
```
./PanzerDiscFE_global_data.exe --test=builder
```

**Pull request merged**